components/eam/src/physics/cam/zm_conv.F90

```diff
@@ -886,8 +1544,152 @@ subroutine zm_convr(lchnk   ,ncol    , &
          dlf (ideep(i),k) = dlg  (i,k)
          pflx(ideep(i),k) = pflxg(i,k)
          ql  (ideep(i),k) = qlg  (i,k)
+
+         sprd(ideep(i),k) = sprdg(i,k)
+         dif (ideep(i),k) = dig  (i,k)
+         dsf (ideep(i),k) = dsg  (i,k)
+         dnlf(ideep(i),k) = dnlg  (i,k)
+         dnif(ideep(i),k) = dnig  (i,k)
+         dnsf(ideep(i),k) = dnsg  (i,k)
+         lambdadpcu(ideep(i),k) = lambdadpcug(i,k)
+         mudpcu(ideep(i),k)     = mudpcug(i,k)
+         frz(ideep(i),k)  = frzg(i,k)*latice/cpres
+         if (zm_microp) then
```

Comment:
Similar to my comment above about moving the allocation and initialization to their own routines, I think having a copy routine to collapse this whole block to a  single line would be a good idea.

Reply:
This block is moved into a subroutine.